### PR TITLE
docs: Use apidoc directive in typed/persistence.md (#22904) (first part)

### PR DESCRIPTION
**Pull request by Captain1653**
References #22904

This file is quite big and it can hard to review it. I supposed to split it.

### CODE REVIEW

File: akka-docs/src/main/paradox/typed/persistence.md [hunk 1]
```diff
@@ -207,8 +207,8 @@ Scala
 Java
 :  @@snip [BasicPersistentBehaviorTest.java](/akka-persistence-typed/src/test/java/jdocs/akka/persistence/typed/BasicPersistentBehaviorTest.java) { #event-handler }
 
-@scala[These are used to create an `EventSourcedBehavior`:]
-@java[These are defined in an `EventSourcedBehavior`:]
+@scala[These are used to create an @scaladoc[EventSourcedBehavior](akka.persistence.typed.scaladsl.EventSourcedBehavior):]
+@java[These are defined in an @javadoc[EventSourcedBehavior](akka.persistence.typed.javadsl.EventSourcedBehavior):]
```
Comment (Captain1653) [inline akka-docs/src/main/paradox/typed/persistence.md:211]: There is class `Effect` in the file. Which link should I use: `akka.persistence.typed.state.javadsl.Effect` or `akka.persistence.typed.javadsl.Effect`? I don't know :(

Comment (raboof) [reply akka-docs/src/main/paradox/typed/persistence.md:211]: `akka.persistence.typed.state.javadsl.Effect` is used for the Durable State feature (https://doc.akka.io/docs/akka/current/typed/index-persistence-durable-state.html). On this page the correct effect type is `akka.persistence.typed.javadsl.Effect`.

**Review by raboof — APPROVED** [review]

**Review by patriknw — APPROVED** [review]
LGTM

**Comment by Captain1653** [comment]
@patriknw  2.6.20 milestone?

**Pull request merged**